core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java

```diff
@@ -165,6 +167,14 @@ public HiveSqlDialect(Context context) {
     case FORMAT:
       unparseFormat(writer, call, leftPrec, rightPrec);
       break;
+    case SYSTEM_FUNCTION:
+      if (call.getOperator().equals(CURRENT_USER)) {
```

Comment:
This seems to be a common code for hive and spark. Move this unparse logic to SqlLibraryOperators

Reply:
done.